helpers/response/query_response.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from graphql import GraphQLError
+from utilities.validations import validate_date_range
+
+
+def filter_rooms_by_responses(
+    Query, info, upper_limit_count, lower_limit_count
+):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for response in all_responses:
+        reponse_count = response.total_responses
+        if lower_limit_count <= reponse_count <= upper_limit_count:
+            filtered_responses.append(response)
+    return filtered_responses
+
+
+def filter_response_by_date(Query, info, upper_date_limit, lower_date_limit):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for responses in all_responses:
+        for response in responses.response:
+            validate_response_dates(
+                response,
+                upper_date_limit, lower_date_limit, filtered_responses
+            )
+        setattr(responses, 'response', filtered_responses)
+        filtered_responses = []
+    return all_responses
+
+
+def validate_response_dates(
+    response, upper_date_limit, lower_date_limit, filtered_responses
+):
+    upper_date_limit = datetime.strptime(upper_date_limit, '%Y %b %d')
+    lower_date_limit = datetime.strptime(lower_date_limit, '%Y %b %d')
+    validate_date_range(
+        upper_date_limit=upper_date_limit, lower_date_limit=lower_date_limit
+    )
+    if (response.created_date < upper_date_limit and
+            response.created_date > lower_date_limit):
+        filtered_responses.append(response)
+    return filtered_responses
+
+
+def check_limits(lower_limit, upper_limit, typ):
+    if (
+      (isinstance(lower_limit, typ)
+          and not isinstance(upper_limit, typ))
+      or (isinstance(upper_limit, typ)
+            and not isinstance(lower_limit, typ))):
+        raise GraphQLError(
+            "Provide upper and lower limits to filter")
+
+
+def check_room_instance(obj, function, **kwargs):
+    if ((isinstance(kwargs['upper_limit'], obj)) and kwargs['upper_limit']
```

Comment:
Kindly write a function that handles the validation of the limits and handles the validation at the entry point.

Reply:
Please add more clarity to this. I do not understand what you mean by this.

Reply:
This has been handled @Bonifase